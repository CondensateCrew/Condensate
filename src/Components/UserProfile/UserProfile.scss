@import '../../css/index.scss';
@import '../../css/variables.scss';

.user-profile-div {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: $primary-gradient;
  border-radius: $one-line-border-radius;
  box-shadow: $external-shadow;
  padding: $one-line-padding-space;
  margin: $small-space 0;
  width: 100%;
  position: relative;
  h2 {
    width: 100%;
    text-align: left;
    margin-left: $small-space;
  }
  img {
    width: $medium-text;
    &.profile-pic {
      width: $large-text;
      border-radius: $round-border-radius;
      padding: $tiny-space;
      box-shadow: $inner-shadow;
    }
  }
  &>section {
    background: $primary-gradient;
    box-shadow: $external-shadow;
    border-radius: $one-line-border-radius;
    display: -webkit-flex;
    display: -ms-flex;
    display: flex;
    position: absolute;
    right: $small-space;
    padding: $small-space;
    img {
      cursor: pointer;
      &:hover {
        opacity: 1;
        transform: scale(1.1);
      }
      &:active {
        transform: scale(0.8);
      }
    }
    .close {
      opacity: 0.4;
      margin-left: $medium-space;
    }
  }
}

.menu-svg {
  width: 1rem;
  &:hover {
    cursor:pointer;
  }
}
