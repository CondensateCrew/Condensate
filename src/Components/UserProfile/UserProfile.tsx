<<<<<<< HEAD
import React from 'react';
import { useSelector } from 'react-redux';
import { AppStore } from 'interfaces';
import AlanBird from '../../assets/avatar.jpg';
import menu from '../../assets/menu-dot.svg';
import './UserProfile.scss';

const UserProfile:React.FC = () => {
  const { firstName, lastName } = useSelector((state:AppStore) => ({
    firstName: state.user.firstName,
    lastName: state.user.lastName
  }))
=======
import './UserProfile.scss';
import React, { useState } from 'react';
import AlanBird from '../../assets/avatar.jpg';
import menu from '../../assets/menu-dot.svg';
import close from '../../assets/close.svg';
import logout from '../../assets/logout.svg';
import { logOutUser } from 'redux/actions';
import { useDispatch } from 'react-redux';

interface Props {
  firstName: string,
  lastName: string,
}

const UserProfile:React.FC<Props> = ({ firstName, lastName }) => {
  const dispatch = useDispatch();
  const [ isClicked, setIsClicked ] = useState<boolean>(false);

  const toggleMenu = (): void => setIsClicked(!isClicked);

  const logoutUser = () => dispatch(logOutUser());
>>>>>>> Refactor UserProfile code

  return (
    <div className='user-profile-div'>
      <img className='profile-pic' src={AlanBird} alt='user profile'/>
      <h2>{`${firstName} ${lastName}`}</h2>
      {!isClicked
        ? <img className='menu-svg' src={menu} alt='menu-svg' onClick={toggleMenu} />
        : <section>
          <img className='logout' src={logout} alt='logout' onClick={logoutUser}/>
          <img className='close' src={close} alt='close' onClick={toggleMenu} />
        </section>
      }
    </div>
  )
}

export default UserProfile
