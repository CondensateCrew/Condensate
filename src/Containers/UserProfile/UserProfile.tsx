import './UserProfile.scss';
import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { useCookies } from 'react-cookie';
import { useDispatch } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { cleanStore } from 'redux/actions';
import { AppStore } from 'interfaces';
import AlanBird from '../../assets/avatar.jpg';
import menu from '../../assets/menu-dot.svg';
import close from '../../assets/close.svg';
import logout from '../../assets/logout.svg';

const UserProfile:React.FC = () => {
  const dispatch = useDispatch();
  const { firstName, lastName } = useSelector((state:AppStore) => ({
    firstName: state.user.firstName,
    lastName: state.user.lastName
  }));
  const [ isClicked, setIsClicked ] = useState<boolean>(false);
  const [ isRedirected, setRedirected ] = useState<boolean>(false);
  const [ cookies, setCookie, removeCookie ] = useCookies(['token', 'name']); //eslint-disable-line

  const toggleMenu = (): void => setIsClicked(!isClicked);

  const logoutUser = () => {
    dispatch(cleanStore());
    removeCookie('name');
    removeCookie('token');
    setRedirected(true);
  };

  return (
    <div className='user-profile-div'>
      { isRedirected && <Redirect to='/' /> }
      <img className='profile-pic' src={AlanBird} alt='user profile'/>
      <h2>{`${firstName || 'Alan'} ${lastName || 'Birds'}`}</h2>
      {!isClicked
        ? <img className='menu-svg' src={menu} alt='menu-svg' onClick={toggleMenu} />
        : <section>
          <img className='logout' src={logout} alt='logout' onClick={logoutUser}/>
          <img className='close' src={close} alt='close' onClick={toggleMenu} />
        </section>
      }
    </div>
  );
}

export default UserProfile;
