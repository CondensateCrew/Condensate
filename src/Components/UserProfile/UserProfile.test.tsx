import React from 'react';
import { shallow, mount } from 'enzyme';
import UserProfile from './UserProfile';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: () => ({
    firstName: 'Alan',
    lastName: 'Birds'
  }),
  useDispatch: () => mockDispatch
}));

describe('UserProfile', () => {
  let wrapper;
  beforeEach(() => {
    interface Props {
      firstName: string,
      lastName: string,
    }
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    const mockFirst = 'Alan';
    const mockLast = 'Bird';
    
    wrapper = shallow(<UserProfile 
      firstName={mockFirst}
      lastName={mockLast}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});

