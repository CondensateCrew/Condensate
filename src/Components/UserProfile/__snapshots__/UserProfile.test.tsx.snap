// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile should match the snapshot 1`] = `
<div
  className="user-profile-div"
>
  <img
    alt="user profile"
    className="profile-pic"
    src="avatar.jpg"
  />
  <h2>
    <span>
      Alan
    </span>
     
    <span>
      Birds
    </span>
  </h2>
  <img
    alt="menu-svg"
    className="menu-svg"
    src="menu-dot.svg"
  />
</div>
`;
